src/validation.cpp

```diff
@@ -3228,9 +3234,20 @@ bool CChainState::AcceptBlockHeader(const CBlockHeader& block, CValidationState&
             return true;
         }
 
-        if (!CheckBlockHeader(block, state))
+        if (!CheckBlockHeader(block, state, aggPubkeys))
             return error("%s: Consensus::CheckBlockHeader: %s, %s", __func__, hash.ToString(), FormatStateMessage(state));
 
+        //if this header was valid and has an aggpubkey change remember it until we finish processing the headers message
+        if(aggPubkeys && (TAPYRUS_XFIELDTYPES)block.xfieldType == TAPYRUS_XFIELDTYPES::AGGPUBKEY
+          && block.xfield.size() == CPubKey::COMPRESSED_PUBLIC_KEY_SIZE
+          && CPubKey(block.xfield.begin(), block.xfield.end()) != aggPubkeys->rbegin()->aggpubkey)
+        {
+            aggPubkeyAndHeight x;
+            x.aggpubkey = CPubKey(block.xfield.begin(), block.xfield.end());
+            x.height = -1;
```

Comment:
Why this height is `-1`?

Reply:
"-1" because we do not expect it to be a valid value in the blockchain.  As this is within headers message processing, we do not know the actual block height. Since this is a temporary list and only the last aggpubkey in the list is always used to check the next block proof, height is unused.

Reply:
in the new commit I moved this code such that the index entry is created and height is a valid value.

Reply:
As a result, can type of height of XFieldAggpubkey also be changed from int32_t to uint32_t, along with `AggPubkeyAndHeight` struct?